### PR TITLE
Change loading text to spinning loading icon

### PR DESCRIPTION
**Pull request by dhstack**
Added a spinning gear when the buttons are "loading" after being clicked instead of the different text.

It makes for a consistent experience across the site.

image
It spins when live

### CODE REVIEW

**Comment by lipis** [comment]
I would add something like that in the `forms.html`

```html
# macro data_loading_text(text='Please wait..', icon='fa fa-spin fa-spinner')
  # if icon
    data-loading-text="<span class='{{icon}}'></span> {{text}}"
  # else
    data-loading-text="{{text}}"
  # endif
# endmacro

```

And then use it like `{{forms.data_loading_text()}}`

p.s. By default I would have the generic text but if someone wants only the spinner could do the: `{{forms.data_loading_text('')}}`

**Comment by dhstack** [comment]
This is a much better way to implement it. I'll work on it.

File: main/templates/macro/forms.html [hunk 1]
```diff
@@ -186,3 +186,12 @@ <h4 class="panel-title">
...
 # endmacro
+
+
+# macro data_loading_text(text='Please wait..', icon='fa fa-spin fa-spinner')
+  # if icon
+    data-loading-text="<span class='{{icon}}'></span> {{text}}"
+  # else
+    data-loading-text="{{text}}"
+  # endif
+# endmacro
```
Comment (lipis) [inline main/templates/macro/forms.html]: end of line

**Comment by lipis** [comment]
why did you close this one?!

**Comment by dhstack** [comment]
I was trying to squash my commits and messed up...

**Comment by dhstack** [comment]
accidentally deleted the branch, but I made a new one with the same name thinking that would work. It didn't.

**Comment by lipis** [comment]
I would recommend wait until all comments remarks are there and then I could just let you know what to type in order to achieve that easily if there are no merges, because you can do that without rebase :)

In short that's one easy way of doing it: http://stackoverflow.com/a/5201642/8418

**Comment by lipis** [comment]
You could also create the same branch name locally and then force push the changes and that would not close the PR..

**Comment by dhstack** [comment]
So that's the approach I tried (reset --soft), but it was telling me I needed to merge when I tried to push to server. Not sure what I did wrong. I'll get it eventually. Thanks for bearing with me until I do though :)

Since I can't reopen this PR, I'll just make a new one.

**Comment by gmist** [comment]
``` shell
git rebase -i master
git push -f
```
imho it is most simple and logical way for squash commits

**Comment by lipis** [comment]
or that... usually @gmist is right :dancer: 
I'm not the guru when it comes to the rebasing/squshing/etc but lately the rebase worked nicely.. :)

**Comment by dhstack** [comment]
@gmist I'll try that next time, thanks.

**Pull request closed without merging**